src/libpmemfile-posix/inode_array.c

```diff
@@ -149,3 +149,82 @@ inode_array_unregister(PMEMfilepool *pfp,
 	mutex_tx_unlock_on_commit(&cur->mtx);
 
 }
+
+/*
+ * inode_array_traverse -- traverses whole inode array and calls specified
+ * callback function for each inode
+ */
+void
+inode_array_traverse(PMEMfilepool *pfp, TOID(struct pmemfile_inode_array) arr,
+		inode_cb inode_cb)
+{
+	struct pmemfile_inode_array *cur = D_RW(arr);
+
+	while (cur) {
+		if (cur->used) {
+			uint32_t used = cur->used;
+			for (unsigned i = 0; used && i < NUMINODES_PER_ENTRY;
+					++i) {
+				if (TOID_IS_NULL(cur->inodes[i]))
+					continue;
+				inode_cb(pfp, cur->inodes[i]);
+				used--;
```

Comment:
How come this routine decreases the `used` field? The `inode_cb` must be `inode_free`? Then there is no point in having an `inode_cb` argument, if that only be one thing, as it sort of is true right now.